src/document.js

```diff
@@ -0,0 +1,254 @@
+      headerLinks.push({
+        href: `${this.props.docUrl}/raw/`,
+        text: `Original file`,
+        icon: 'fa fa-cloud-download'
+      })
+    }
+
+    let ocrData = Object.keys((data.ocr || {})).map((tag, index) => {
+      return {tag: tag, text: data.ocr[tag]}
+    })
+    headerLinks.push(...ocrData.map(({tag}) => {
+      return {
+        href: `${this.props.docUrl}/ocr/${tag}/`,
+        text: `OCR  ${tag}`,
+        icon: 'fa fa-cloud-download'
+      }
+    }))
+
+    return (
+      <div className="card doc-page">
+
+        <div className="lead">#{doc.id}: {data.filename}</div>
+        <DocumentHeaderLinks data={headerLinks} />
+
+        <div className="bg-faded doc-section-title">Meta</div>
+
+        <table className="table table-sm">
+          <tbody>
+            <tr>
+              <td className="col">Path</td>
+              <td><code>{data.path}</code></td>
+            </tr>
+            <tr>
+              <td>Filename</td>
+              <td><code>{data.filename}</code></td>
+            </tr>
+            {data.type &&
+              <tr>
+                <td>Type</td>
+                <td><code>{data.type}</code></td>
+              </tr>
+            }
+            {data.type != 'folder' && data.md5 &&
+              <tr>
+                <td>MD5</td>
+                <td><code>{data.md5}</code></td>
+              </tr>
+            }
+            {data.type != 'folder' && data.sha1 &&
+              <tr>
+                <td>SHA1</td>
+                <td><code>{data.sha1}</code></td>
+              </tr>
+            }
+            {data.lang &&
+              <tr>
+                <td>Language</td>
+                <td><code>{data.lang}</code></td>
+              </tr>
+            }
+            {data['date-created'] &&
+              <tr>
+                <td>Created</td>
+                <td>{data['date-created']}</td>
+              </tr>
+            }
+            {data.date &&
+              <tr>
+                <td>Modified</td>
+                <td>{data.date}</td>
+              </tr>
+            }
+            {data.pgp &&
+              <tr>
+                <td>PGP</td>
+                <td>{data.pgp}</td>
+              </tr>
+            }
+          </tbody>
+        </table>
+
+        <DocumentEmailSection doc={doc} />
+        <DocumentFilesSection title="Files"
+                              data={files} baseUrl={this.baseUrl} />
+        <DocumentTextSection title="Text"
+                             text={doc.content.text} />
+        <DocumentTextSection title="Headers &amp; Parts"
+                             text={doc.content.tree} />
+        {ocrData.map(({tag, text}) =>
+          <DocumentTextSection title={tag} text={text} />
+        )}
+      </div>
+    )
+  }
+
+}
+
+class DocumentEmailSection extends React.Component {
+
+  render() {
+    let doc = this.props.doc || {}
+    let data = doc.content
+    let files = doc.children || []
+
+    if(data.type == 'email') {
```

Comment:
It would make more sense to check for `data.type == 'email'` outside this component, in `Document.render`.